subsys/net/ip/dhcpv4.c

```diff
@@ -1236,6 +1273,35 @@ static void dhcpv4_start_internal(struct net_if *iface, bool first_start)
 	k_mutex_unlock(&lock);
 }
 
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS)
+
+int net_dhcpv4_add_option_callback(struct net_dhcpv4_option_callback *cb)
+{
+	__ASSERT(cb, "No callback!");
+	__ASSERT(cb->handler, "No callback handler!");
```

Comment:
It's a public API, so it'd be better if the function returned an error instead of asserting.